stdlib/builtins.pyi

```diff
@@ -109,19 +112,19 @@ class object:
     def __dir__(self) -> Iterable[str]: ...
     def __init_subclass__(cls) -> None: ...
 
-class staticmethod(object):  # Special, only valid as a decorator.
-    __func__: Callable[..., Any]
+class staticmethod(Generic[_FuncT]):  # Special, only valid as a decorator.
+    __func__: _FuncT
     __isabstractmethod__: bool
-    def __init__(self, f: Callable[..., Any]) -> None: ...
+    def __init__(self, f: _FuncT) -> None: ...
     def __new__(cls: Type[_T], *args: Any, **kwargs: Any) -> _T: ...
-    def __get__(self, obj: _T, type: Optional[Type[_T]] = ...) -> Callable[..., Any]: ...
+    def __get__(self, obj: _T, type: Type[_T] | None = ...) -> _FuncT: ...
 
-class classmethod(object):  # Special, only valid as a decorator.
-    __func__: Callable[..., Any]
+class classmethod(Generic[_ClsT, _P, _T_co]):  # Special, only valid as a decorator.
+    __func__: Callable[Concatenate[_ClsT, _P], _T_co]
     __isabstractmethod__: bool
-    def __init__(self, f: Callable[..., Any]) -> None: ...
+    def __init__(self, f: Callable[Concatenate[_ClsT, _P], _T_co]) -> None: ...
```

Comment:
Perhaps we could use self types to get rid of the missing generics error. If I have understood it correctly, that is a valid way to provide the generic parameters.

```suggestion
    def __init__(self: classmethod[_ClsT, _P, _T_co], f: Callable[Concatenate[_ClsT, _P], _T_co]) -> None: ...
```